### PR TITLE
Add mingw to dyn_mex.m and document compiler options and installation.

### PR DESCRIPTION
**Pull request by JohannesPfeifer**
Related to #1226

### CODE REVIEW

File: doc/dynare.texi [hunk 1]
```diff
@@ -554,6 +544,59 @@ Then you will need to recompile the pre-processor and the dynamic
...
+For older version of MATLAB, in particular before R2014a, it may sometimes make sense to use the gcc compiler 
+provided by Cygwin. However, integrating it in MATLAB can be quite cumbersome and should be considered as a
+legacy option. For details, see
+@uref{http://www.dynare.org/DynareWiki/ConfigureMatlabWindowsForMexCompilation,instructions
+on the Dynare wiki}. 
+
+@node On Debian GNU/Linux and Ubuntu
+@subsection On Debian GNU/Linux and Ubuntu
+
+Users of MATLAB under Linux normally need to do nothing,
```
Comment (houtanb) [inline doc/dynare.texi]: This is not true. Debian users must install the gcc compilers via `apt-get install build-essential`

File: matlab/utilities/general/dyn_mex.m [hunk 2]
```diff
@@ -52,7 +52,10 @@ function dyn_mex(win_compiler,basename,force)
           % Add /TP flag as fix for #1227
           eval(['mex -O LINKFLAGS="$LINKFLAGS /export:Dynamic" COMPFLAGS="/TP" ' basename '_dynamic.c ' basename '_dynamic_mex.c'])
           eval(['mex -O LINKFLAGS="$LINKFLAGS /export:Static" COMPFLAGS="/TP" ' basename '_static.c ' basename '_static_mex.c'])
-      elseif strcmp(win_compiler,'cygwin')
+      elseif strcmp(win_compiler,'mingw')
+          eval(['mex -O LINKFLAGS="$LINKFLAGS /export:Dynamic" ' basename '_dynamic.c ' basename '_dynamic_mex.c'])
+          eval(['mex -O LINKFLAGS="$LINKFLAGS /export:Static"  ' basename '_static.c ' basename '_static_mex.c'])
```
Comment (houtanb) [inline matlab/utilities/general/dyn_mex.m]: please remove the whitespace at the end of this line

File: doc/dynare.texi (hunk 1 above)
Comment (JohannesPfeifer) [inline doc/dynare.texi]: @houtanb That part was already in the manual. I will adjust it accordingly

**Comment by JohannesPfeifer** [comment]
@houtanb Updated.

File: matlab/utilities/general/dyn_mex.m [hunk 3]
```diff
@@ -62,7 +65,7 @@ function dyn_mex(win_compiler,basename,force)
                 basename '_static.c ' basename '_static_mex.c'])
       else
         error(['When using the USE_DLL option, you must give either ' ...
-               '''cygwin'' or ''msvc'' option to the ''dynare'' command'])
+               '''cygwin'', ''minggw'' or ''msvc'' option to the ''dynare'' command'])
```
Comment (houtanb) [inline matlab/utilities/general/dyn_mex.m]: there is a typo here. `mingw` not `minggw`

Comment (JohannesPfeifer) [inline matlab/utilities/general/dyn_mex.m]: @ That was stupid. I updated again.

**Pull request merged**